typescript/infra/scripts/utils.ts

```diff
@@ -78,6 +79,14 @@ export function withContext<T>(args: yargs.Argv<T>) {
     .demandOption('context');
 }
 
+export function withProtocol<T>(args: yargs.Argv<T>) {
+  return args
+    .describe('protocol', 'protocol type')
+    .default('protocol', ProtocolType.Ethereum)
+    .coerce('protocol', coerceProtocol)
```

Comment:
I think you can do `.choices` or `.options` or something with an enum to avoid this coercion stuff